### DISCUSSION TITLE
Missing feature when locking days

### DISCUSSION THREAD

**Discussion by stell** [easepick/easepick#74] [Ideas]
The Lock Plugin can disable days for selection. When inseparable is active, people cannot choose a range over a disabled date, which is good.
What's not possible (as far I can see) is disabling days for selection (no arrival dates) but still making it possible to choose a range over them when inseparable is active.
So there need to be two different locked states. One only disabled and one completely blocked/booked without the possibility of selecting a range over when inseparable is active.

**Comment by nathanmedhurstschoolshr** [comment]
if you disable inseperable you can book over the locked days, however it does still count the locked day, if anyone has a way of bypassing this?

**Comment by fredasterisk** [comment]
I'm looking for exactly this!
Fully booked/blocked -> inseparable : true
Disabled or no checkin/checkout -> inseparable : false

Any clues?

**Comment by 1ucay** [comment]
I need inseparable: true, but allow to choose days over the locked weekend :)) 
What we need is add another filter function "filterRange". Now we can control start and stop range over locked days.

**Reply by 1ucay** [reply]
Well it was simple to solved. Now you can select range over locked weekend :)

	LockPlugin: {
		minDate: false,
		inseparable: true,
		filter(date, picked, event) {
			if ( el.id === 'rent-date-format' ) {

				if ( date.inArray(bookedDates, '[)') )
					return true;

				return false;

			}
			return date;
		}
	},
	setup(picker) {

		picker.on('view', (e) => {

			const { view, date, target } = e.detail;

			if ( 'CalendarDay' === view ) {
				if ( date ) {
					if ( [0, 6].includes(date.getDay()) ) {
						target.classList.add('locked');
					}
				}
			}

		})
